.github/workflows/snapshot-release.yml

```
@@ -63,7 +63,13 @@ jobs:
           ref: ${{ steps.refs.outputs.head_ref }}
           fetch-depth: 0
 
-      - run: git fetch origin main:main
+      - name: Extract base branch from .changeset/config.json
+        id: getBaseBranch
+        run: |
+          baseBranch=$(jq -r '.baseBranch' .changeset/config.json)
+          echo "baseBranch=${baseBranch}" >> $GITHUB_ENV
+
+      - run: git fetch origin ${{ env.baseBranch }}:${{ env.baseBranch }}
```

Maybe we should use the output instead to avoid setting the global env vars.
```suggestion
          echo "baseBranch=${baseBranch}" >> $GITHUB_OUTPUT

      - run: git fetch origin ${{ steps.getBaseBranch.outputs.baseBranch }}:${{  steps.getBaseBranch.outputs.baseBranch }}
```